File: sdk/runanywhere-swift/Sources/RunAnywhere/Public/Models/FrameworkAvailability.swift
import Foundation

/// Detailed information about framework availability
public struct FrameworkAvailability {
    /// The framework being described
    public let framework: LLMFramework

    /// Whether this framework is available (has a registered adapter)
    public let isAvailable: Bool

    /// Reason why the framework is not available (if applicable)
    public let unavailabilityReason: String?

    /// Hardware requirements for optimal performance
    public let requirements: [HardwareRequirement]

    /// Recommended use cases for this framework
    public let recommendedFor: [String]

    /// Model formats supported by this framework
    public let supportedFormats: [ModelFormat]

    public init(
        framework: LLMFramework,
        isAvailable: Bool,
        unavailabilityReason: String? = nil,
        requirements: [HardwareRequirement] = [],
        recommendedFor: [String] = [],
        supportedFormats: [ModelFormat] = []
    ) {
        self.framework = framework
        self.isAvailable = isAvailable
        self.unavailabilityReason = unavailabilityReason
        self.requirements = requirements
        self.recommendedFor = recommendedFor
        self.supportedFormats = supportedFormats
    }
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/models/FrameworkAvailability.kt
package com.runanywhere.sdk.models

/**
 * Detailed information about framework availability
 */
data class FrameworkAvailability(
    /**
     * The framework being described
     */
    val framework: LLMFramework,
    
    /**
     * Whether this framework is available (has a registered adapter)
     */
    val isAvailable: Boolean,
    
    /**
     * Reason why the framework is not available (if applicable)
     */
    val unavailabilityReason: String? = null,
    
    /**
     * Hardware requirements for optimal performance
     */
    val requirements: List<HardwareRequirement> = emptyList(),
    
    /**
     * Recommended use cases for this framework
     */
    val recommendedFor: List<String> = emptyList(),
    
    /**
     * Model formats supported by this framework
     */
    val supportedFormats: List<ModelFormat> = emptyList()
) 